uwblueprint/project-read-frontend
# add test build and test job to sequential CI flow

### Notion ticket link



[Ensure automated tests pass CI](https://www.notion.so/uwblueprintexecs/Ensure-automated-tests-pass-in-CI-c883376b4076478790cbf97cd9269718)



### Implementation description

- Create a build job and a test job
- Run them sequentially, test job runs after build job passes
- Refer to [this template](https://circleci.com/docs/2.0/sample-config/#sequential-workflow) of a sequential workflow
- Add dependency caching, example [here](https://circleci.com/docs/2.0/yarn/#caching)
- Add workspace to save project state between jobs. Concept diagram [here](https://circleci.com/docs/2.0/concepts/#caches-workspaces-and-artifacts)
- Add environment variable under circleci project settings. [See here for reference](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project)





### What should reviewers focus on?

- Good CI practices

### Checklist
